ambry-frontend/src/main/java/com/github/ambry/frontend/AmbrySecurityService.java

```diff
@@ -126,6 +130,13 @@ public void postProcessRequest(RestRequest restRequest, Callback<Void> callback)
       } else if (storageQuotaService != null && storageQuotaService.shouldThrottle(restRequest)) {
         exception = new RestServiceException("StorageQuotaExceeded", RestServiceErrorCode.TooManyRequests);
       } else {
+        if (quotaManager != null) {
+          ThrottlingRecommendation throttlingRecommendation = quotaManager.getThrottleRecommendation(restRequest);
+          if (throttlingRecommendation != null && throttlingRecommendation.shouldThrottle()) {
+            // TODO Use ThrottlingRecommendation to set response headers about quota usage.
+            exception = new RestServiceException("User Quota Exceeded", RestServiceErrorCode.TooManyRequests);
```

Comment:
should throw this exception, if you only set the exception, then code in the rest of this block would still get executed.